playbooks/openshift-glusterfs/config.yml

```diff
@@ -1,10 +1,15 @@
 ---
 - import_playbook: ../init/main.yml
   vars:
-    l_init_fact_hosts: "oo_masters_to_config:oo_nodes_to_config:oo_glusterfs_to_config"
+    # We need facts on all nodes since we need to check (on RPM installs) that
+    # glusterfs-fuse is available across the cluster.
+    l_init_fact_hosts: "oo_masters_to_config:oo_nodes_to_config"
+    # We only want sanity and base packages on OpenShift nodes that will be
+    # GlusterFS nodes, as they may be new nodes to the OpenShift cluster.
+    l_glusterfs_nodes: "{{ groups['oo_nodes_to_config'] | intersect(groups['oo_glusterfs_to_config']) }}"
     l_openshift_version_set_hosts: "oo_masters_to_config:!oo_first_master"
-    l_sanity_check_hosts: "{{ groups['oo_masters_to_config'] | union(groups['oo_glusterfs_to_config']) }}"
+    l_sanity_check_hosts: "{{ groups['oo_masters_to_config'] | union(l_glusterfs_nodes) }}"
     l_install_base_packages: True
-    l_base_packages_hosts: "oo_nodes_to_configi:&oo_glusterfs_to_config"
```

Comment:
There's no reason to change this.  If they are gluster hosts and oo_nodes, they need to be native.  I don't think running an 'external' gluster on node hosts should be supported.

Reply:
The reason I do this is for the default of `['none]'`. I want the `none` here so that base packages doesn't run for any external GlusterFS node, but I DON'T want the `none` in the line above because something like `master:none` ends up skipping the sanity check on `master`.

Reply:
oo_nodes_to_config:$oo_glusterfs_to_config will only run on hosts that are both nodes and gluster.

There is a typo in the existing, 'oo_nodes_to_configi' should be 'oo_nodes_to_config'

"none" is not valid for group names and doesn't work as you would expect.  I don't have a problem with any other lines, just this one.

Reply:
The problem I ran into is that I just wanted to reuse `l_glusterfs_nodes` here, but since it's a hosts list I needed to join all `l_glusterfs_nodes` hosts into a string. When that string is empty, the default is to run on `oo_all_hosts`. Is there anything I can do to keep the intended functionality, or am I only left with using `oo_nodes_to_config:$oo_glusterfs_to_config` again?

Reply:
I don't know why oo_nodes_to_config:&oo_glusterfs_to_config would be insufficient for what you're trying to do, other than say 'it's possible to have external gluster hosts also be nodes' in which case I think that's a terrible idea, generally.

Reply:
I'm not saying it isn't, I'm saying I want to reuse the variable `l_glusterfs_nodes` if I can, given that its contents are functionally equivalent to the alternative string. Is there any way to do that?

Reply:
Ideally, we could use our special grouping to do something like oo_glusterfs_native and oo_glusterfs_external or something like that and totally refactor the glusterfs, but that would be a giant refactor.

Yeah, some things are lists of hosts, and some things are groups, I prefer groups over lists of hosts wherever possible.